### ISSUE TITLE
AppCompat 23.2.0

### ISSUE THREAD

**Issue by marbat87** [gabrielemariotti/changeloglib#28]
Ciao,

Aggiornando le librerie Appcompat etc di Google alla versione 23.2.0, le voci del changelog appaiono sparse a grande distanza una dall'altra.

Il mio codice è questo:
<pre>
&lt;it.gmariotti.changelibs.library.view.ChangeLogRecyclerView
            android:layout_width="match_parent"
            android:layout_height="match_parent"
            risuscito:changeLogFileResourceId="@raw/changelog_full"
            risuscito:rowHeaderLayoutId="@layout/changelogrowheader_material_layout"
            risuscito:rowLayoutId="@layout/changelogrow_material_layout"
            android:id="@+id/aboutText" />
</pre>

Grazie in anticipo.

**Comment by gabrielemariotti** [comment]
Grazie del feedback. Controllerò cosa è cambiato con la 23.2.0

**Comment by Zeliret** [comment]
Probably you should use wrap_content instead of match_parent in the header and item layout. 

> This release brings an exciting new feature to the LayoutManager API: auto-measurement! This allows a RecyclerView to size itself based on the size of its contents.

For me there is another problem. I have a crash: 

```java
java.lang.NullPointerException: Attempt to read from field 'int android.support.v7.widget.RecyclerView$ItemAnimator$ItemHolderInfo.left' on a null object reference
                                                                 at android.support.v7.widget.SimpleItemAnimator.animateDisappearance(SimpleItemAnimator.java:78)
                                                                 at android.support.v7.widget.RecyclerView.animateDisappearance(RecyclerView.java:3246)
                                                                 at android.support.v7.widget.RecyclerView.access$700(RecyclerView.java:147)
                                                                 at android.support.v7.widget.RecyclerView$4.processDisappeared(RecyclerView.java:422)
                                                                 at android.support.v7.widget.ViewInfoStore.process(ViewInfoStore.java:231)
                                                                 at android.support.v7.widget.RecyclerView.dispatchLayoutStep3(RecyclerView.java:3086)
                                                                 at android.support.v7.widget.RecyclerView.dispatchLayout(RecyclerView.java:2914)
                                                                 at android.support.v7.widget.RecyclerView.onLayout(RecyclerView.java:3277)
                                                                 at android.view.View.layout(View.java:16636)
                                                                 at android.view.ViewGroup.layout(ViewGroup.java:5437)
                                                                 at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
                                                                 at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
                                                                 at android.view.View.layout(View.java:16636)
                                                                 at android.view.ViewGroup.layout(ViewGroup.java:5437)
                                                                 at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
                                                                 at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
                                                                 at android.view.View.layout(View.java:16636)
                                                                 at android.view.ViewGroup.layout(ViewGroup.java:5437)
                                                                 at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1743)
                                                                 at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1586)
                                                                 at android.widget.LinearLayout.onLayout(LinearLayout.java:1495)
                                                                 at android.view.View.layout(View.java:16636)
                                                                 at android.view.ViewGroup.layout(ViewGroup.java:5437)
                                                                 at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
                                                                 at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
                                                                 at android.view.View.layout(View.java:16636)
                                                                 at android.view.ViewGroup.layout(ViewGroup.java:5437)
                                                                 at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
                                                                 at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
                                                                 at android.view.View.layout(View.java:16636)
                                                                 at android.view.ViewGroup.layout(ViewGroup.java:5437)
                                                                 at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
                                                                 at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
                                                                 at android.view.View.layout(View.java:16636)
                                                                 at android.view.ViewGroup.layout(ViewGroup.java:5437)
                                                                 at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1743)
                                                                 at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1586)
                                                                 at android.widget.LinearLayout.onLayout(LinearLayout.java:1495)
                                                                 at android.view.View.layout(View.java:16636)
                                                                 at android.view.ViewGroup.layout(ViewGroup.java:5437)
                                                                 at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
                                                                 at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
                                                                 at com.android.internal.policy.PhoneWindow$DecorView.onLayout(PhoneWindow.java:2678)
                                                                 at android.view.View.layout(View.java:16636)
                                                                 at android.view.ViewGroup.layout(ViewGroup.java:5437)
                                                                 at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:2171)
                                                                 at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1931)
                                                                 at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1107)
                                                                 at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6013)
                                                                 at android.view.Choreographer$CallbackRecord.run(Choreographer.java:858)
                                                                 at android.view.Choreographer.doCallbacks(Choreographer.java:670)
                                                                 at android.view.Choreographer.doFrame(Choreographer.java:606)
                                                                 at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:844)
                                                                 at android.os.Handler.handleCallback(Handler.java:739)
                                                                 at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                 at android.os.Looper.loop(Looper.java:148)
                                                                 at android.app.ActivityThread.main(ActivityThread.java:5417)
                                                                 at java.lang.reflect.Method.invoke(Native Method)
                                                                 at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
                                                             	at com.android.internal.os.ZygoteInit.main(ZygoteInit.jav
```

My layout:
```xml
<it.gmariotti.changelibs.library.view.ChangeLogRecyclerView
    android:id="@+id/view"
    xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    android:clipToPadding="false"
    android:paddingBottom="@dimen/padding_narrow"
    android:paddingTop="@dimen/padding_narrow"
    app:rowHeaderLayoutId="@layout/header_changelog"
    app:rowLayoutId="@layout/item_changelog"/>
```

**Comment by marbat87** [comment]
I created a pull request with the fix for the layout issue and with updates for libraries and plugins: #30

Regards

**Comment by marbat87** [comment]
Hi @gabrielemariotti, any news?

**Comment by gabrielemariotti** [comment]
@marbat87 Thanks for the code.
I have to check and to find the time to build the new version of the library.

**Comment by gabrielemariotti** [comment]
@marbat87  The item space issue is fixed with #83626cb

**Comment by gabrielemariotti** [comment]
@Zeliret I open a new issue on your stacktrace.Please refer to #31

**Comment by gabrielemariotti** [comment]
While I am trying to fix the #31 I've fix the 23.2.0 issue with a snapshot:

You can now download it from Sonatype's snapshots repo.

1. Add the snapshots repo to your `build.gradle`

    ```groovy
    repositories {
        maven { url "https://oss.sonatype.org/content/repositories/snapshots/" }
    }
    ```

2. Add build dependency

    ```groovy
    dependencies {
        compile 'com.github.gabrielemariotti.changeloglib:changelog:2.0.1-SNAPSHOT@aar'
    }
    ```

**Comment by mtrakal** [comment]
Thanks, `2.0.1-SNAPSHOT` looks that working for me (first look). I can later finally confirm after release to betatest.

**Comment by gabrielemariotti** [comment]
Fixed with the 2.1.0
